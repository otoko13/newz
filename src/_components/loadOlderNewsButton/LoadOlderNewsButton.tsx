import React from 'react';
import './loadOlderNewsButton.scss';
import { ActionButton } from 'office-ui-fabric-react/lib/Button';
import { CARD_BOX_HEIGHT } from '../itemCard/ItemCard';

export interface ILoadOlderNewsButtonProps {
    numberOfItemsLoaded: number;
    onLoadOlderNewsClicked: () => void;
}

const LoadOlderNewsButton = (props: ILoadOlderNewsButtonProps) => {
    return (
        <div className='LoadOlderNewsButton' style={{top: `${CARD_BOX_HEIGHT * props.numberOfItemsLoaded}px` }}>
            <ActionButton onClick={props.onLoadOlderNewsClicked} text='Click to see older news' />'
        </div>
    );
}

export default LoadOlderNewsButton;