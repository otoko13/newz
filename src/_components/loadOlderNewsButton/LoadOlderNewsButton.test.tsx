import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';
import LoadOlderNewsButton, { ILoadOlderNewsButtonProps } from './LoadOlderNewsButton';

let props: ILoadOlderNewsButtonProps;

beforeEach(() => {
    props = {
        onLoadOlderNewsClicked: jest.fn(),
    }
});

describe('snapshot', () => {
    it('should look sensible', () => {
        const { container } = render(<LoadOlderNewsButton {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});

describe('clicking button', () => {
    it('should call props.onLoadOlderNewsClicked', () => {
        const { getByText } = render(<LoadOlderNewsButton {...props} />);
        expect(props.onLoadOlderNewsClicked).not.toHaveBeenCalled();
        fireEvent.click(getByText('Click to see older news'));
        expect(props.onLoadOlderNewsClicked).toHaveBeenCalled();
    });
});