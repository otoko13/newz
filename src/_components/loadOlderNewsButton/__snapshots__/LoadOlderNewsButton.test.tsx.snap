// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should look sensible 1`] = `
<div
  class="LoadOlderNewsButton"
>
  <button
    class="ms-Button ms-Button--action ms-Button--command root-40"
    data-is-focusable="true"
    type="button"
  >
    <span
      class="ms-Button-flexContainer flexContainer-41"
      data-automationid="splitbuttonprimary"
    >
      <span
        class="ms-Button-textContainer textContainer-42"
      >
        <span
          class="ms-Button-label label-44"
          id="id__0"
        >
          Click to see older news
        </span>
      </span>
    </span>
  </button>
  '
</div>
`;
